src/Client.php

```diff
@@ -182,6 +182,13 @@ public function send(RequestInterface $request)
                 return new FutureResponse(new RejectedPromise($e));
             }
             throw RequestException::wrapException($trans->request, $e);
+        } catch (\TypeError $error) {
```

Comment:
I did it to satisfy [ClientTest::testEnsuresResponseIsPresentAfterSending](https://github.com/guzzle/guzzle/blob/3b9826a2/tests/ClientTest.php#L392), which failed with

> 1) GuzzleHttp\Tests\ClientTest::testEnsuresResponseIsPresentAfterSending
Failed asserting that exception of type "TypeError" matches expected exception "\GuzzleHttp\Exception\RequestException".

On second thought that is probably a flawed test. 

The PHP docs state that "Error is the base class for all internal PHP errors." The subclasses are DivisionByZeroError, AssertionError, ParseError, and TypeError. All of those indicate incorrect code, not exceptional runtime behaviour (and are certainly not RequestExceptions). Converting them to Exceptions changes the semantics. Error and Exception are in different type hierarchies so they can be handled differently.

So even though it is wrong, wrapping TypeError is required to stay backwards compatible. Wrapping all Errors would be more wrong.